### ISSUE TITLE
TextInPut MaxLength

### ISSUE THREAD

**Issue by peng-oss** [facebook/react-native#32732]
### Description

for example : 

MaxLength is 200 

199 Text +  1 😈 emoji   textValue clear

### Version

"react-native": "0.65.1",

### Output of `react-native info`

o

### Steps to reproduce

0

**Comment by Shahanshah87** [comment]
Hey, did you find any solution?

**Comment by peng-oss** [comment]
> Hey, did you find any solution?

no,it's react-native bug

**Comment by gitBingxu** [comment]
This issue may have been resolved in [v 0.66.0](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v0660). 

My solution in react-native@v0.63:
```
import React, { forwardRef, memo, useRef } from 'react'
import { TextInput, TextInputProps } from 'react-native'
import { useMemoizedFn } from 'ahooks'

export interface MaxLenInputProps extends Omit<TextInputProps, 'maxLength' | 'onChange'> {
    /*
     * 允许输入内容的最大长度，必填
     */
    maxLength: number
}
const MaxLenInputCompo = forwardRef<TextInput, MaxLenInputProps>((props, ref) => {
    const {
        value, onKeyPress, onChangeText, maxLength
    } = props

    const notChangeRef = useRef(false)

    const originalOnKeyPress = useMemoizedFn((...args: Parameters<typeof onKeyPress>) => {
        const [event] = args
        const { nativeEvent: { key } } = event
        const valueLen = value?.length || 0
        const keyLen = key.length

        const isEmoji = keyLen > 1 && !['Backspace', 'Enter'].includes(key)
        if (isEmoji) {
            const hasEnoughLen = maxLength - valueLen >= keyLen
            if (!hasEnoughLen) {
                // 如果输入的字符是 emoji，并且没有足够的长度容纳该字符，那么我们需要舍弃本次输入的字符。
                notChangeRef.current = true
            }
        }

        onKeyPress?.(...args)
    })

    const originalOnChangeText = useMemoizedFn((...args: Parameters<typeof onChangeText>) => {
        const [text] = args
        const emitValue = notChangeRef.current ? value : text

        onChangeText?.apply?.(null, [emitValue, ...args.slice(1)])
        notChangeRef.current = false
    })

    return (
        <TextInput
            ref={ref}
            {...props}
            value={value}
            maxLength={maxLength}
            onKeyPress={originalOnKeyPress}
            onChangeText={originalOnChangeText} />
    )
})

export const MaxLenInput = memo(MaxLenInputCompo)
```
